### PR TITLE
Forget Profunctor property tests

### PR DESCRIPTION
**Pull request by CristhianMotoche**
Hi! This should help in part to solve #7

I found these property tests really similar to the ones of `(->)`

Please, let me know if you have any suggestions or comments.

### CODE REVIEW

File: test/Test/Prolens/Property.hs [hunk 1]
```diff
@@ -61,6 +61,20 @@ profunctorsSpec = describe "Profunctor" $ do
...
+            dimap id id f x === f x
+        it "Composition: dimap (ab . bc) (yz . xy) ≡ dimap bc yz . dimap ab xy" $ hedgehog $ do
+            f  <- unForget <$> forAllWith (const "f")  genForget
+            ab <- unForget <$> forAllWith (const "ab") genForget
+            bc <- unForget <$> forAllWith (const "bc") genForget
+            xy <- unForget <$> forAllWith (const "xy") genForget
+            yz <- unForget <$> forAllWith (const "xy") genForget
+
+            n <- forAll genInt
+            dimap (ab . bc) (yz . xy) f n === (dimap bc yz . dimap ab xy) f n
```
Comment (chshersh) [inline test/Test/Prolens/Property.hs]: It looks like this property doesn't test the instance for `Forget`, but it still tests the instance for function. Your property generates `Forget` but then unwraps it and gets an ordinary function. So, here actually the `Profunctor` instance for `(->)` is tested.

I think you need to write a function `eqForget` similar to `eqFun` below, unwrap `Forget` inside that function and use it here.

Let us know if you need any help with this :slightly_smiling_face:

**Review by chshersh — COMMENTED** [review]
Hi @CristhianMotoche! Thanks for helping with this :muscle: 

I left one comment on how to make properties more robust, but the general idea is correct :+1:

File: test/Test/Prolens/Property.hs (hunk 1 above)
Comment (CristhianMotoche) [reply test/Test/Prolens/Property.hs]: Thanks @chshersh I knew I was missing something! Thank you for the guidance. Please, let me know if the property tests look better now.

File: test/Test/Prolens/Property.hs [hunk 2]
```diff
@@ -61,8 +61,27 @@ profunctorsSpec = describe "Profunctor" $ do
...
+        it "Composition: dimap (ab . bc) (yz . xy) ≡ dimap bc yz . dimap ab xy" $ hedgehog $ do
+            f  <- forAllWith (const "f")  genForget
+            ab <- forAllWith (const "ab") genFunction
+            bc <- forAllWith (const "bc") genFunction
+            xy <- forAllWith (const "xy") genFunction
+            yz <- forAllWith (const "xy") genFunction
+
+            eqForget
+                (dimap (ab . bc) (yz . xy) f)
+                (((dimap bc yz) . (dimap ab xy)) f)
```
Comment (vrom911) [inline test/Test/Prolens/Property.hs]: You can reduce the amount of parenthesis to make it clearer 🙂 
```suggestion
                ((dimap bc yz . dimap ab xy) f)
```

**Review by vrom911 — COMMENTED** [review]
Looks great! Thanks a lot for your work, @CristhianMotoche 💯

File: test/Test/Prolens/Property.hs (hunk 2 above)
Comment (CristhianMotoche) [reply test/Test/Prolens/Property.hs]: Sure thing @vrom911

**Review by chshersh — APPROVED** [review]
Looks great, @CristhianMotoche! :+1:

**Pull request merged**